src/main/java/io/temporal/internal/testservice/StateMachines.java

```diff
@@ -980,15 +975,19 @@ private static void scheduleActivityTask(
       ScheduleActivityTaskCommandAttributes d,
       long workflowTaskCompletedEventId) {
     TestServiceRetryState retryState;
-    if (d.hasRetryPolicy()) {
-      RetryPolicy retryPolicy = d.getRetryPolicy();
-      long expirationInterval = TimeUnit.SECONDS.toMillis(d.getScheduleToCloseTimeoutSeconds());
-      long expirationTime = data.store.currentTimeMillis() + expirationInterval;
-      retryState = new TestServiceRetryState(retryPolicy, expirationTime);
+    RetryPolicy retryPolicy;
+    if (!d.hasRetryPolicy()) {
+      // If no activity retry policy is supplied, assume a default to mimic a standard Temporal
+      // server
+      retryPolicy = toRetryPolicy(RetryOptions.newBuilder().validateBuildWithDefaults()).build();
```

Comment:
We should compy defaults from the service and do not rely on validateBuildWithDefaults which was intended for the SDK only validation.